src/app/downloads/downloads-chart.component.ts

```diff
@@ -98,27 +99,42 @@ export class DownloadsChartComponent implements OnDestroy {
       // no partial date range coverage charts, makes the loading UX too jerky
       const expectedLength = dateUtil.getAmountOfIntervals(this.filter.beginDate, this.filter.endDate, this.filter.interval);
-      if (this.podcastChartData && this.podcastChartData.data.length === expectedLength &&
-        (this.episodeChartData && this.episodeChartData.length > 0 &&
-        this.episodeChartData.every(chartData => chartData.data.length === expectedLength))) {
-        // if we have episodes to combine with podcast total
-        const episodeDatasets = this.episodeChartData.map(m => m.data);
-        const allOtherEpisodesData: TimeseriesChartModel = {
-          data: subtractTimeseriesDatasets(this.podcastChartData.data, episodeDatasets),
-          label: 'All Other Episodes',
-          color: neutralColor
-        };
-        this.chartData = [...this.episodeChartData, allOtherEpisodesData];
-      } else if (this.podcastMetrics && this.podcastMetrics.charted && this.podcastChartData && this.podcastChartData.data.length > 0) {
-        this.chartData = [this.podcastChartData];
-      } else if (this.episodeChartData && this.episodeChartData.length > 0 &&
-        this.episodeChartData.every(chartData => chartData.data.length === expectedLength)) {
-        this.chartData = this.episodeChartData;
-      } else {
-        this.chartData = null;
+      switch (this.filter.chartType) {
+        case CHARTTYPE_STACKED:
+          if (this.podcastChartData && this.podcastMetrics.charted && this.podcastChartData.data.length === expectedLength &&
+            (this.episodeChartData && this.episodeChartData.length > 0 &&
+            this.episodeChartData.every(chartData => chartData.data.length === expectedLength))) {
+            // if we have episodes to combine with podcast total
+            const episodeDatasets = this.episodeChartData.map(m => m.data);
+            const allOtherEpisodesData: TimeseriesChartModel = {
+              data: subtractTimeseriesDatasets(this.podcastChartData.data, episodeDatasets),
+              label: 'All Other Episodes',
+              color: neutralColor
+            };
+            this.chartData = [...this.episodeChartData, allOtherEpisodesData];
+          } else if (this.podcastChartData && this.podcastMetrics.charted &&
+            this.podcastChartData.data.length > 0 && this.episodeChartData.length === 0) {
+            this.chartData = [this.podcastChartData];
+          } else if (this.episodeChartData && this.episodeChartData.length > 0 &&
+            this.episodeChartData.every(chartData => chartData.data.length === expectedLength)) {
+            this.chartData = this.episodeChartData;
+          }
+          break;
+        case CHARTTYPE_PODCAST:
+          if (this.podcastMetrics && this.podcastChartData && this.podcastChartData.data.length > 0) {
+            this.chartData = [this.podcastChartData];
+          }
+          break;
+        case CHARTTYPE_EPISODES:
```

Comment:
I find myself wishing the episode chart also had a line for the podcast-total.  But that's just my user-feedback, not a requirement.  (And the way you have also makes logical sense to me).